internal/controller/cilium_flow_collector.go

```diff
@@ -28,32 +29,63 @@ const (
 	ciliumHubbleRelayNamespace string = "kube-system"
 )
 
-// newCiliumFlowCollector connects to Cilium Hubble Relay, sets up an Observer client, and returns a new Collector using it.
-func newCiliumFlowCollector(ctx context.Context, logger *zap.Logger, ciliumNamespace string, tlsAuthProperties tls.AuthProperties) (*CiliumFlowCollector, error) {
+// newCiliumFlowCollector connects to Cilium Hubble Relay, sets up an Observer client,
+// and returns a new Collector using it. It tries namespaces until discovery succeeds.
+func newCiliumFlowCollector(ctx context.Context, logger *zap.Logger, ciliumNamespaces []string, tlsAuthProperties tls.AuthProperties) (*CiliumFlowCollector, error) {
 	clientset, err := NewClientSet()
 	if err != nil {
 		return nil, fmt.Errorf("failed to create new client set: %w", err)
 	}
+	var hubbleAddress string
+	var tlsConfig *cryptoTls.Config
+	var discoveryErr error
 
-	service, err := hubble.DiscoverCiliumHubbleRelay(ctx, ciliumNamespace, clientset)
-	if err != nil {
-		return nil, fmt.Errorf("failed to discover Cilium Hubble Relay service: %w", err)
-	}
+	for _, ciliumNamespace := range ciliumNamespaces {
+		// Step 1: Try to discover Hubble Relay
+		service, err := hubble.DiscoverCiliumHubbleRelay(ctx, ciliumNamespace, clientset)
+		if err != nil {
+			logger.Warn("Failed to discover Cilium Hubble Relay service",
```

Comment:
On GKE, this would just be noise at Warn level. Lower to Debug.
```suggestion
			logger.Debug("Failed to discover Cilium Hubble Relay service",
```